src/hooks/pagination.rs

```diff
@@ -211,6 +212,31 @@ where
     }
 }
 
+/// Apply pagination state to a set of data.
+///
+/// Ideally, pagination is applied on the source of the data. Like a database query. However,
+/// sometimes it can be convenient to even paginate a fully loaded dataset.
+///
+/// This hook takes a full dataset and returns the currently selected page. It will update
+/// whenever the entries or pagination control state changes.
+#[hook]
+fn use_apply_pagination<T>(entries: Rc<Vec<T>>, control: PaginationControl) -> Rc<Vec<T>>
```

Comment:
Could we accept `impl Iterator` as the input? Then it could be used on filtered datasets and would only lazily collect once in this function. The code below is generic iterator code anyway.

Reply:
Yea, that might get trick. As `use_memo` requires something to hold on to, passing in only a reference to the function. And that thing also needs to be `PartialEq`. Not sure there are many iterators which would implement this.

Reply:
I see, didn't think of the PartialEq. Bummer. Then I would say we can merge it like this :+1: